ygot/struct_validation_map_test.go

```diff
@@ -407,3 +407,264 @@ func TestInitContainer(t *testing.T) {
 		}
 	}
 }
+
+func TestMergeJSON(t *testing.T) {
+	tests := []struct {
+		name    string
+		inA     map[string]interface{}
+		inB     map[string]interface{}
+		want    map[string]interface{}
+		wantErr bool
+	}{{
+		name: "simple maps",
+		inA:  map[string]interface{}{"a": 1},
+		inB:  map[string]interface{}{"b": 2},
+		want: map[string]interface{}{"a": 1, "b": 2},
+	}, {
+		name: "non-overlapping multi-layer tree",
+		inA: map[string]interface{}{
+			"a": map[string]interface{}{
+				"a1": 42,
+			},
+			"aa": map[string]interface{}{
+				"aa2": 84,
+			},
+		},
+		inB: map[string]interface{}{
+			"b": map[string]interface{}{
+				"b1": 42,
+			},
+			"bb": map[string]interface{}{
+				"bb2": 84,
+			},
+		},
+		want: map[string]interface{}{
+			"a": map[string]interface{}{
+				"a1": 42,
+			},
+			"aa": map[string]interface{}{
+				"aa2": 84,
+			},
+			"b": map[string]interface{}{
+				"b1": 42,
+			},
+			"bb": map[string]interface{}{
+				"bb2": 84,
+			},
+		},
+	}, {
+		name: "overlapping trees",
+		inA: map[string]interface{}{
+			"a": map[string]interface{}{
+				"b": "c",
+			},
+		},
+		inB: map[string]interface{}{
+			"a": map[string]interface{}{
+				"c": "d",
+			},
+		},
+		want: map[string]interface{}{
+			"a": map[string]interface{}{
+				"b": "c",
+				"c": "d",
+			},
+		},
+	}, {
```

Comment:
Maybe add a test case where one tree is deeper than the other.

Reply:
Done.